File: lib/layer2_core/light/e_primary_fixed_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EPrimaryFixedLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl20;
      case 'standard': return cstl20;
      case 'blue': return cbl20;
      case 'green': return cgl20;
      case 'red': return crl20;
      case 'nothing': return cnol20;
      default: return cstl20;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EPrimaryFixedLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EPrimaryFixedLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl20
            AppTheme.STANDARD -> R.color.cstl20
            AppTheme.BLUE -> R.color.cbl20
            AppTheme.GREEN -> R.color.cgl20
            AppTheme.RED -> R.color.crl20
            AppTheme.NOTHING -> R.color.cnol20
        }
    }
}
